book/src/lib/rpc/mod.rs

```diff
@@ -51,3 +51,46 @@ pub async fn rpc_server(
 
     Ok((server, addr))
 }
+
+#[cfg(test)]
+mod tests {
+    use std::net::TcpListener;
+
+    use helpers::result::StdResult;
+
+    use super::*;
+    use crate::db;
+
+    const DATABASE_URL: &str = "postgres://postgres:password@localhost";
+
+    #[tokio::test]
+    async fn rpc_server_binding_should_work() -> StdResult<()> {
+        // Arrange
+        let db_pool = db::get_db_pool(DATABASE_URL);
+
+        // Act
+        let _ = rpc_server(&addr(), db_pool).await?;
+
+        // Assert
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn rpc_server_binding_should_fail() {
+        // Arrange
+        let socket = TcpListener::bind::<SocketAddr>(addr()).unwrap();
+        let addr = socket.local_addr().unwrap();
+
+        let db_pool = db::get_db_pool(DATABASE_URL);
+
+        // Act
+        let _ = rpc_server(&addr, db_pool).await.unwrap();
+
+        // Assert
+    }
+
+    fn addr() -> SocketAddr {
+        SocketAddr::from(([127, 0, 0, 1], 0))
+    }
```

Comment:
This is testing if the standard library correctly gets a network port assigned and then if tokio blocks the reuse of this port. It basically tests if the standard library, tokio and the operating system work correctly. Is it really necessary?

Reply:
Yeaah, I see your point, but I want to test that `fn rpc_server` actually binds that address. I theory it could also succeed even if the address is in use.